backup-stores/gcs/src/main/java/io/camunda/zeebe/backup/gcs/manifest/Manifest.java

```diff
@@ -11,18 +11,53 @@
 
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import io.camunda.zeebe.backup.api.Backup;
+import io.camunda.zeebe.backup.api.BackupStatus;
+import io.camunda.zeebe.backup.api.BackupStatusCode;
 import io.camunda.zeebe.backup.common.BackupDescriptorImpl;
 import io.camunda.zeebe.backup.common.BackupIdentifierImpl;
+import io.camunda.zeebe.backup.common.BackupStatusImpl;
 import java.time.Instant;
+import java.util.Optional;
 
 @JsonSerialize(as = ManifestImpl.class)
 @JsonDeserialize(as = ManifestImpl.class)
 public sealed interface Manifest {
 
-  static InProgressManifest createManifest(
-      final BackupIdentifierImpl id, final BackupDescriptorImpl descriptor) {
+  static InProgressManifest create(Backup backup) {
     final var creationTime = Instant.now();
-    return new ManifestImpl(id, descriptor, IN_PROGRESS, creationTime, creationTime);
+    return new ManifestImpl(
+        BackupIdentifierImpl.from(backup.id()),
+        BackupDescriptorImpl.from(backup.descriptor()),
+        IN_PROGRESS,
+        creationTime,
+        creationTime);
+  }
+
+  static BackupStatus toStatus(Manifest manifest) {
+    return switch (manifest.statusCode()) {
+      case IN_PROGRESS -> new BackupStatusImpl(
+          manifest.id(),
+          Optional.ofNullable(manifest.descriptor()),
+          BackupStatusCode.IN_PROGRESS,
+          Optional.empty(),
+          Optional.ofNullable(manifest.createdAt()),
+          Optional.ofNullable(manifest.modifiedAt()));
+      case COMPLETED -> new BackupStatusImpl(
+          manifest.id(),
+          Optional.ofNullable(manifest.descriptor()),
+          BackupStatusCode.COMPLETED,
+          Optional.empty(),
+          Optional.ofNullable(manifest.createdAt()),
+          Optional.ofNullable(manifest.modifiedAt()));
+      case FAILED -> new BackupStatusImpl(
+          manifest.id(),
+          Optional.ofNullable(manifest.descriptor()),
+          BackupStatusCode.FAILED,
+          Optional.ofNullable(manifest.asFailed().failureReason()),
+          Optional.ofNullable(manifest.createdAt()),
+          Optional.ofNullable(manifest.modifiedAt()));
+    };
```

Comment:
:wrench: :bulb: We could think about whether Manifest implements BackupStatus, so we can simply return manifest instead of converting.